frontend/src/_components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -0,0 +1,202 @@
+              <div className="d-flex flex-column">
+                <span style={{ width: 'auto' }} className="fw-400 fs-12">
+                  Set to null
+                </span>
+              </div>
+              <div>
+                <label className={`form-switch`}>
+                  <input
+                    className="form-check-input"
+                    type="checkbox"
+                    // checked={nullValue}
+                    // onChange={() => setNullValue((prev) => !prev)}
+                  />
+                </label>
+              </div>
+            </div>
+          )}
+
+          {defaultValue !== null && (
+            <div className="d-flex align-items-center gap-2">
+              <div className="d-flex flex-column">
+                <span style={{ width: 'auto' }} className="fw-400 fs-12">
+                  Set to default
+                </span>
+              </div>
+              <div>
+                <label className={`form-switch`}>
+                  <input
+                    className="form-check-input"
+                    type="checkbox"
+                    // checked={defaultValue}
+                    // onChange={() => handleDefaultChange(columnDetails?.column_default, !defaultValue)}
+                  />
+                </label>
+              </div>
+            </div>
+          )}
+        </div>
+      </div>
+    );
+  };
+
+  const SaveChangesFooter = () => {
+    return (
+      <div className={cx('d-flex align-items-center', 'justify-content-end')}>
+        <div className="d-flex" style={{ gap: '8px' }}>
+          <ButtonSolid onClick={handleCancel} variant="tertiary" size="sm" className="fs-12 p-2">
+            Cancel
+          </ButtonSolid>
+          <ButtonSolid
+            onClick={handleSave}
+            // disabled={cellValue == previousCellValue ? true : false}
+            variant="primary"
+            size="sm"
+            className="fs-12 p-2"
+          >
+            Save
+          </ButtonSolid>
+        </div>
+      </div>
+    );
+  };
+
+  const CustomCalendarContainer = ({ children }) => {
+    return (
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'center',
+          width: '280px',
+          borderRadius: '6px',
+          boxShadow: '0px 8px 16px 0px #3032331A',
+        }}
+      >
+        <div style={{ width: '100%' }}>{children}</div>
+        <div
+          className={`d-flex flex-column gap-3`}
+          style={{
+            width: '100%',
+            padding: '12px',
+            marginBottom: '12px',
+            borderTop: '1px solid var(--Slate-05, #E6E8EB)',
+          }}
+        >
+          <SaveChangesSection />
+          <SaveChangesFooter />
+        </div>
+      </div>
+    );
+  };
+
+  const memoizedCustomCalendarContainer = useCallback(CustomCalendarContainer, []);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/ToolJet/ToolJet/pull/10066#issuecomment-2163100421>#2c1b8e</a><span> - 06/12/2024, 02:29 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'useCallback' hook is used incorrectly for 'CustomCalendarContainer'. This can lead to unnecessary re-renders and performance issues because 'useCallback' should be used for functions, not components.
</div>
<div id="fix">
<b>Fix:</b> Use 'useMemo' instead of 'useCallback' to memoize the 'CustomCalendarContainer' component.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div>
<pre><code>
 @@ -144,7 +144,7 @@
 
  const memoizedCustomCalendarContainer = useCallback(CustomCalendarContainer, []);
 +const memoizedCustomCalendarContainer = useMemo(() => CustomCalendarContainer, []);
 
  const darkMode = localStorage.getItem('darkMode') === 'true';
</code></pre>
</div>
</details>
</div>
<br>


</div>